### PR TITLE
2579 Design lookup table API using OpenAPI spec

### PR DESCRIPTION
**Pull request by carlosfelix2**
This PR adds the design for the lookup table API.

**Background**
We wish to allow secure access to manipulate the lookup tables that are stored in the database.  Authenticated users will be able to manage the lookup tables and create new versions to quickly respond to data requirements.  This API will serve for a possible future UI.  This API will also be used by the PRIME CLI management commands.

**Goals**
An API that is useful for a future management UI page.
An API that is useful for on-boarding states by the ReportStream team.
An API which doesn't require specific code knowledge.
An API which allows the separation of operations by sensitivity
An API which is implementable in an Azure Function.
Require Okta authentication to use all endpoints.

**API**
A set of REST functions to manage the lookup tables as follows:
**OUTDATED - SEE OPEN API DOC**

_Fetch the list lookup tables_
HTTP Method: GET
Return: JSON array with list of table information
HTTP Status Code: 200 for successful fetch, 400 if there was an error
api/lookuptables - fetch a list of active tables
api/lookuptables?showAll=true - fetch all table including inactive

_Fetch the contents of a lookup table_
HTTP Method: GET
Return: JSON array with list of table rows
HTTP Status Code: 200 for successful fetch, 404 if the table name was not found, 400 if there was an error
api/lookuptables/<table-name> - Fetch the latest version of a table
api/lookuptables/<table-name>/<version-number> - Fetch the given version of a table

_Create a new version of a lookup table_
HTTP Method: POST
Payload: JSON array with list of table rows
Return: JSON object with new table version information
HTTP Status Code: 201 for successful creation, 400 if there was an error
api/lookuptables/<table-name>

_Activate a specific version of a table_
HTTP Method: POST
Payload: None
Return: JSON object with activated table version information
HTTP Status Code: 200 for successful activation, 404 if the table was not found, 400 if there was an error
api/lookuptables/<table-name>/<version>

## Changes
- Added lookup table API OpenAPI spec

### Testing
- [ ] Tested locally?
- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] Downloaded a file from `http://localhost:7071/api/download`?
- [ ] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Fixes
- #2579

### CODE REVIEW

File: prime-router/docs/api/lookup-tables-openapi.yaml [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+              schema:
+                $ref: '#/components/schemas/ErrorMsg'
+        '400':
+          description: if there was an error fetching the table data
+          content:
+            text/plain:
+              schema:
+                $ref: '#/components/schemas/ErrorMsg'
+    post:
+      description: Activate a specific version of a table
```
Comment (jimduff-usds) [inline prime-router/docs/api/lookup-tables-openapi.yaml:106]: How does deactivation of a table version occur?
Is there a rule that only one version of a table can be active at a time?
If so, does activating a version also deactivate all other versions?

Comment (carlosfelix2) [reply prime-router/docs/api/lookup-tables-openapi.yaml:106]: yes, only one version at a time is allowed, and it is enforced via a database constraint.  In the code on the other PR you can see how I deactivate and activate within the same transaction.

Comment (RickHawesUSDS) [reply prime-router/docs/api/lookup-tables-openapi.yaml:106]: What is the content for this post?

File: prime-router/docs/api/lookup-tables-openapi.yaml [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorMsg'
+        '400':
+          description: if there was an error fetching the table data
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorMsg'
+    post:
```
Comment (RickHawesUSDS) [inline prime-router/docs/api/lookup-tables-openapi.yaml:105]: I suggest that document that post requires PRIME admin rights, while get is open.

**Review by clediggins-usds — COMMENTED** [review]
Is the intent that every minor change to any setting will result in a new version of that settings table?  For example, a receiver's status changes from 'inactive' to 'testing' and then to 'active' - is that 3 different versions?  If not, we may want a way to make minor updates/spelling fixes at `PATCH /api/lookuptables/{tableName}/{tableVersion}`.

File: prime-router/docs/api/lookup-tables-openapi.yaml (hunk 1 above)
Comment (RickHawesUSDS) [reply prime-router/docs/api/lookup-tables-openapi.yaml:106]: This is really a REST-style comment. Reading REST posts should have a representation of the state change that you want to effect. In other words, you should have a content part that represents the `isActive` state.

File: prime-router/docs/api/lookup-tables-openapi.yaml [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+              schema:
+                $ref: '#/components/schemas/TableVersionInfo'
+        '400':
+          description: if there was an error fetching the list of tables
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorMsg'
+
+  /api/lookuptables/{tableName}:
```
Comment (whytheplatypus) [inline prime-router/docs/api/lookup-tables-openapi.yaml:37]: Given the behavior of `/api/lookuptables` and `/api/lookuptables/{tableName}/{tableVersion}` I'd expect this to return a list of versions for this specific table. Is the intent for this to return the `latest` version of the table or the `active` version of a table? Endpoints that are aliases for each might be needed.

Comment (carlosfelix2) [reply prime-router/docs/api/lookup-tables-openapi.yaml:37]: We only want to return the active table data.  I can see how we can benefit from a better endpoint here.

Comment (carlosfelix2) [reply prime-router/docs/api/lookup-tables-openapi.yaml:37]: I added /content.  Take a look again

**Pull request merged**